test/accum.c

```diff
@@ -2219,56 +2227,94 @@ test_swmr_write_big(hbool_t newest_format)
     if (HDmemcmp(wbuf2, rbuf, (size_t)BIG_BUF_SIZE) != 0)
         TEST_ERROR;
 
-    /* Fork child process to verify that the data at [1024, 2014] does get written to disk */
-    if ((pid = HDfork()) < 0) {
-        HDperror("fork");
-        FAIL_STACK_ERROR;
-    }
-    else if (0 == pid) { /* Child process */
-        /* By convention, argv[0] tells the name of program invoked.
-         *
-         * execv on NetBSD 8 will actually return EFAULT if there is a
-         * NULL at argv[0], so we follow the convention unconditionally.
-         */
-        char        swmr_reader[] = SWMR_READER;
-        char *const new_argv[]    = {swmr_reader, NULL};
-        /* Run the reader */
-        status = HDexecv(SWMR_READER, new_argv);
-        HDprintf("errno from execv = %s\n", HDstrerror(errno));
-        FAIL_STACK_ERROR;
-    } /* end if */
+#if defined(H5_HAVE_WIN32_API)
+    {
+        STARTUPINFO         si;
+        PROCESS_INFORMATION pi;
+        DWORD               exit_code = EXIT_FAILURE;
 
-    /* Parent process -- wait for the child process to complete */
-    while (pid != HDwaitpid(pid, &status, 0))
-        /*void*/;
+        ZeroMemory(&si, sizeof(si));
+        si.cb = sizeof(si);
+        ZeroMemory(&pi, sizeof(pi));
 
-    /* Check if child process terminates normally and its return value */
-    if (WIFEXITED(status) && !WEXITSTATUS(status)) {
-        /* Flush the accumulator */
-        if (accum_reset(rf) < 0)
+        if (0 == CreateProcess(NULL, SWMR_READER, NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi)) {
+            HDprintf("CreateProcess failed (%d).\n", GetLastError());
             FAIL_STACK_ERROR;
+        }
 
-        /* Close and remove the file */
-        if (H5Fclose(fid) < 0)
-            FAIL_STACK_ERROR;
+        (void)WaitForSingleObject(pi.hProcess, INFINITE);
```

Comment:
The return type of WaitForSingleObject() tells you why the wait stopped. We don't care about this and instead check the return value of the sub-program launched by CreateProcess().